docs/modules/ROOT/pages/access-control.adoc

```diff
@@ -202,3 +134,143 @@ TIP: A recommended configuration is to grant both roles to a secure governance c
+
+In order to restrict access to some functions of your contract, you should inherit from the xref:api:access.adoc#AccessManaged[`AccessManaged`] contract provided along with the manager so that the `restricted` modifier is available to protect any externally facing function. Note you will have to set the address of the Access Manager instance (xref:api:access.adoc#AccessManaged-authority--[authority]) in the constructor so the `restricted` modifier can check permissions by calling the correct manager.
+
+Here's a simple example of an xref:tokens.adoc#ERC20[`ERC20`] token that defines a `mint` functionality that is restricted by an xref:api:access.adoc#AccessManager[`AccessManager`]:
+
+```solidity
+include::api:example$access-control/AccessManagedERC20MintBase.sol[]
+```
+
+NOTE: Make sure you fully understand how xref:api:access.adoc#AccessManager[`AccessManager`] works before using it or copy-pasting the examples from this guide.
+
+Once the managed contract has been deployed, it is now under the manager's control, which will get the minter role assigned to an address and also allow the role to call the `mint` function.
+
+```javascript
+const MINTER = 42n; // Roles are uint64 (0 is reserved for the ADMIN_ROLE)
+
+// Allow the minter role to call the function selector
+// corresponding to the mint function
+await manager.setTargetFunctionRole(
+    myToken.address,
+    ['0x40c10f19'], // bytes4(keccak256('mint(address,uint256)'))
+    MINTER,
+    { from: initialAdmin }
+);
+```
+
+Even though each role has its own list of function permissions, each role member (`address`) will have an execution delay that will dictate how long the account should wait to execute a function that requires its role. Delayed operations must be xref:api:access.adoc#AccessManager-schedule-address-bytes-uint48-[`schedule`d] first on the Access Manager before they can be executed, either directly or using the Access Manager's xref:api:access.adoc#AccessManager-execute-address-bytes-[`execute`] function.
+
+Additionally, roles can have a granting delay that prevents adding members immediately. The Access Manager admins can set this grant delay as follows:
+
+```javascript
+const HOUR = 60 * 60;
+
+const GRANT_DELAY = 24 * HOUR;
+const EXECUTION_DELAY = 5 * HOUR;
+    
+await manager.setGrantDelay(MINTER, GRANT_DELAY, { from: initialAdmin });
+await manager.grantRole(MINTER, EXECUTION_DELAY, { from: initialAdmin });
+```
+
+You may notice roles do not define a name. Opposed to the xref:api:access.adoc#AccessControl[`AccessControl`] case, roles aren't hardcoded in the contract as `bytes32` values but identified as numeric values. However, role labeling allows tooling discoverability for easier role exploration and it can be achieved using the xref:api:access.adoc#AccessManager-labelRole-uint64-string-[`labelRole`] function.
```

Comment:
```suggestion
You may notice roles do not define a name. As opposed to the xref:api:access.adoc#AccessControl[`AccessControl`] case, roles aren't hardcoded in the contract as `bytes32` values but identified as numeric values. However, role labeling allows tooling discoverability for easier role exploration and it can be achieved using the xref:api:access.adoc#AccessManager-labelRole-uint64-string-[`labelRole`] function.
```